### ISSUE TITLE
Running on Raspi 4 (headless raspberry pi os (debian)) tries to build core midi

### ISSUE THREAD

**Issue by thejohncotton** [haubie/midiex#2]
I'm so excited to find this library. Hoping to possibly add some contributions to documentation while testing. First thing I noticed was that you need to have the rust tools installed before you can run Mix.install(), once that dependency is handled, and a few other system packages have been installed it seems to work as prescribed. The errors are really nice from rustler.

In addition to running on a mac, I also want to run on a headless pi 4 but am hitting some issues when compiling.

```elixir
iex -S mix
Erlang/OTP 26 [erts-14.0.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [jit]

==> midiex
Compiling 10 files (.ex)
Compiling crate midiex in release mode (native/midiex)
   Compiling memchr v2.5.0
   Compiling proc-macro2 v1.0.63
   Compiling libc v0.2.138
   Compiling quote v1.0.28
   Compiling regex-syntax v0.6.28
   Compiling unicode-ident v1.0.5
   Compiling pkg-config v0.3.26
   Compiling aho-corasick v0.7.20
   Compiling alsa-sys v0.3.1
Compiling lib/midiex/backend.ex (it's taking more than 10s)
   Compiling bitflags v1.3.2
   Compiling void v1.0.2
   Compiling cfg-if v1.0.0
   Compiling syn v2.0.22
   Compiling unreachable v1.0.0
   Compiling heck v0.4.0
   Compiling rustler v0.29.0
   Compiling lazy_static v1.4.0
   Compiling regex v1.7.0
   Compiling nix v0.24.3
   Compiling alsa v0.7.0
   Compiling rustler_sys v2.3.0
   Compiling midir v0.9.1
   Compiling rustler_codegen v0.29.0
   Compiling midiex v0.1.0 (/home/pi/test/deps/midiex/native/midiex)
error[E0433]: failed to resolve: use of undeclared crate or module `coremidi`
  --> src/lib.rs:11:5
   |
11 | use coremidi::Notification::{ObjectAdded, ObjectRemoved};
   |     ^^^^^^^^ use of undeclared crate or module `coremidi`

error[E0432]: unresolved import `coremidi`
  --> src/lib.rs:10:5
   |
10 | use coremidi::{Client, Notification, AddedRemovedInfo, ObjectType};
   |     ^^^^^^^^ use of undeclared crate or module `coremidi`

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `midiex` (lib) due to 2 previous errors

== Compilation error in file lib/midiex/backend.ex ==
** (RuntimeError) Rust NIF compile error (rustc exit code 101)
    (rustler 0.26.0) lib/rustler/compiler.ex:41: Rustler.Compiler.compile_crate/2
    lib/midiex/backend.ex:6: (module)
could not compile dependency :midiex, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile midiex", update it with "mix deps.update midiex" or clean it with "mix deps.clean midiex"
```

I believe that I should be able to not compile coremidi on a non mac platform and compile some other resource, but am unsure how to make the change.

Thanks again! 🎹 ❤️

**Comment by haubie** [comment]
Thanks @thejohncotton,

You are correct in that coremidi shouldn't pull in if its not being compiled on MacOS.

I've taken a look and it looks like I missed a couple of lines where I should have tagged coreaudio as a MacOS-only dependency.

I've made that change now and I've just pushed v0.5.0 to hex with the changes and hopefully that resolves it for you. Let me know how you go.

I'm currently starting the migration to Rustler precompiled, so I'm hoping some of the rust toolchain issues will disappear and it will turn into a simple download for most people.

**Comment by haubie** [comment]
Another little update @thejohncotton,

I've pushed another release v0.5.3 to fix a couple of other errors I noticed being thrown on the rust build side on Ubuntu. Just letting you know incase you run into the same problems on raspberry pi os.

One thing I noticed that even though I had ALSA installed, I needed to have a couple of extra packages for the compilation to complete: `sudo apt install libasound2-dev pkg-config`.

The following threads from another project helped me: https://github.com/Spotifyd/spotifyd/issues/659
